File: app/src/main/java/com/leafchild/scopequotas/data/WorklogType.java
package com.leafchild.scopequotas.data;

/**
 * Created by: leafchild
 * Date: 10/04/2017
 * Project: ScopeQuotas
 */

public enum WorklogType {

	MINUTES("Minutes"),
	HOURS("Hours"),
	DAYS("Days");

	private String value;

	WorklogType(String value) {

		this.value = value;
	}

	public String getValue() {

		return value;
	}

	public static WorklogType fromString(String text) {

		for (WorklogType b : WorklogType.values()) {
			if (b.value.equalsIgnoreCase(text)) {
				return b;
			}
		}
		return null;
	}

	public static WorklogType fromOrdinal(int n) {

		return values()[n];
	}
}

File: app/src/main/java/com/leafchild/scopequotas/data/WorklogType.kt
package com.leafchild.scopequotas.data

/**
 * Created by: leafchild
 * Date: 10/04/2017
 * Project: ScopeQuotas
 */

enum class WorklogType constructor(val value: String) {

    MINUTES("Minutes"),
    HOURS("Hours"),
    DAYS("Days");

    companion object {

        fun fromString(text: String): WorklogType? {

            for (b in WorklogType.values()) {
                if (b.value.equals(text, ignoreCase = true)) {
                    return b
                }
            }
            return null
        }

        fun fromOrdinal(n: Int): WorklogType {

            return values()[n]
        }
    }
}
